src/components/SignIn/index.tsx

```diff
@@ -34,154 +38,177 @@ export interface SignInProps {
-    }
+const SignInComponent = React.memo((props: SignInProps) => {
+    const isMobileDevice = useSelector(selectMobileDeviceState);
+    const history = useHistory();
+    const intl = useIntl();
+    const isValidForm = () => {
+        const isEmailValid = props.email.match(EMAIL_REGEX);
 
-    private handleChangeEmail = (value: string) => {
-        this.props.changeEmail(value);
+        return props.email && isEmailValid && props.password;
     };
 
-    private handleChangePassword = (value: string) => {
-        this.props.changePassword(value);
-    };
+    const handleChangeEmail = (value: string) => {
+            props.changeEmail(value);
+        };
 
-    private handleFieldFocus = (field: string) => {
-        this.props.handleChangeFocusField(field);
-    };
+    const handleChangePassword = (value: string) => {
+            props.changePassword(value);
+        };
 
-    private handleSubmitForm = () => {
-        this.props.refreshError();
-        this.props.onSignIn();
-    };
+    const handleFieldFocus = (field: string) => {
+            props.handleChangeFocusField(field);
+        };
 
-    private isValidForm = () => {
-        const { email, password } = this.props;
-        const isEmailValid = email.match(EMAIL_REGEX);
+    const handleSubmitForm = () => {
+            props.refreshError();
+            props.onSignIn();
+        };
 
-        return email && isEmailValid && password;
-    };
+    const handleValidateForm = () => {
+            props.isFormValid();
+        };
 
-    private handleValidateForm = () => {
-        this.props.isFormValid();
-    };
+    const handleClick = (label?: string, e?: React.FormEvent<HTMLInputElement>) => {
+            if (e) {
+                e.preventDefault();
+            }
+            if (!isValidForm()) {
+                handleValidateForm();
+            } else {
+                handleSubmitForm();
+            }
+        };
 
-    private handleClick = (label?: string, e?: React.FormEvent<HTMLInputElement>) => {
-        if (e) {
-            e.preventDefault();
-        }
-        if (!this.isValidForm()) {
-            this.handleValidateForm();
-        } else {
-            this.handleSubmitForm();
-        }
-    };
-
-    private handleEnterPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    const handleEnterPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
         if (event.key === 'Enter') {
             event.preventDefault();
 
-            this.handleClick();
+            handleClick();
         }
     };
-}
+
+    const renderForgotButton = () => {
+        return <div className="cr-sign-in-form__bottom-section">
+            <div
+                className="cr-sign-in-form__bottom-section-password"
+                onClick={() => props.onForgotPassword(email)}
+            >
+                {props.forgotPasswordLabel || 'Forgot your password?'}
+            </div>
+        </div>;
+    };
+
+    const renderRegister = () => {
```

Comment:
use `useCallback` for all inner methods inside React.FC (component with hooks)

Reply:
use useCallback for ALL inner methods inside React.FC (component with hooks). Because, each render component will create that methods again and again.
This is required for all components.

Reply:
done

Reply:
don't see changes for this

Reply:
now fixed